test/es6/toPrimitive.js

```diff
@@ -44,8 +44,33 @@ var tests = [
             assert.isFalse(o.writable,     "Symbol @@toPrimitive is not writable");
             assert.isFalse(o.enumerable,   "Symbol @@toPrimitive is not enumerable");
             assert.isTrue(o.configurable,  "Symbol @@toPrimitive is configurable");
-            assert.throws(function() { Symbol.prototype[Symbol.toPrimitive](); }, TypeError, "Symbol[@@toPrimitive] throws no matter the parameters", "Symbol[Symbol.toPrimitive]: invalid argument");
-       }
+            assert.throws(function() { Symbol.prototype[Symbol.toPrimitive](); }, TypeError, "Symbol[@@toPrimitive] throws no matter the parameters", "Symbol[Symbol.toPrimitive]: 'this' is not a Symbol object");
+
+            var s = Symbol();
+            assert.areEqual(s, Object(s)[Symbol.toPrimitive](), ""); // true
+            assert.areEqual(s, Symbol.prototype[Symbol.toPrimitive].call(s), ""); // true
+
+            assert.areEqual(Symbol.toPrimitive, Symbol.toPrimitive[Symbol.toPrimitive](), "Symbol.toPrimitive");
+            assert.areEqual(Symbol.iterator, Symbol.iterator[Symbol.toPrimitive](), "Symbol.iterator");
+            assert.areEqual(Symbol.hasInstance, Symbol.hasInstance[Symbol.toPrimitive](), "Symbol.hasInstance");
+            assert.areEqual(Symbol.isConcatSpreadable, Symbol.isConcatSpreadable[Symbol.toPrimitive](), "Symbol.isConcatSpreadable");
+            assert.areEqual(Symbol.match, Symbol.match[Symbol.toPrimitive](), "Symbol.match");
+            assert.areEqual(Symbol.replace, Symbol.replace[Symbol.toPrimitive](), "Symbol.replace");
+            assert.areEqual(Symbol.search, Symbol.search[Symbol.toPrimitive](), "Symbol.search");
+            assert.areEqual(Symbol.split, Symbol.split[Symbol.toPrimitive](), "Symbol.split");
+            assert.areEqual(Symbol.toStringTag, Symbol.toStringTag[Symbol.toPrimitive](), "Symbol.toStringTag");
+            assert.areEqual(Symbol.unscopables, Symbol.unscopables[Symbol.toPrimitive](), "Symbol.unscopables");
+
+            assert.throws(()=>Symbol.prototype[Symbol.toPrimitive].call(undefined), TypeError, "undefined", "Symbol[Symbol.toPrimitive]: 'this' is not a Symbol object");

```

Comment:
True - removed in latest rev.

---
In reply to: [71949683](https://github.com/Microsoft/ChakraCore/pull/1319#discussion_r71949683) [](ancestors = 71949683)